lib/pluginmanager/package.rb

```diff
@@ -0,0 +1,60 @@
+require "logstash/util/compress"
+require "fileutils"
+
+class LogStash::PluginManager::Package < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "compress package as a tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "compress package as a zip file", :default => LogStash::Environment.windows?
+  option "--clean", :flag, "clean up the generated dump of plugins", :default => false
+  option "--override", :flag, "Override a previously generated package file", :default => false
+
+  def execute
+    puts("Packaging plugins for offline usage")
+
+    validate_target_file
+    LogStash::Bundler.invoke!({:package => true, :all => true})
+    compress_cache(LogStash::Environment::CACHE_PATH)
+    FileUtils.rm_rf(LogStash::Environment::CACHE_PATH) if clean?
+
+    puts("Generated at #{target_file}")
+  end
+
+  private
+
+  def compress_cache(location)
+    if zip?
+      LogStash::Util::Zip.compress(location, target_file)
+    else
+      LogStash::Util::Tar.compress(location, target_file)
+    end
+  end
+
+
+  def delete_target_file?
+    return true if override?
+    puts("File #{target_file} exist, do you want to override it to continue? (Y/N)")
+    ( "y" == STDIN.gets.strip.downcase ? true : false)
+  end
+
+
+  def validate_target_file
+    if file.exist?(target_file)
+      if  delete_target_file?
+        File.delete(target_file)
+      else
+        signal_error("Package creation cancelled, a previously generated package exist at location: #{target_file}, move this file to safe place and run the command again")
+        exit
+      end
+    end
+  end
+
+  def target_file
+    target_file = File.join(LogStash::Environment::LOGSTASH_HOME, "plugins_bundle")
```

Comment:
since we updated all working, we should probably also name this "plugins_package" ?